core/trino-main/src/main/java/io/trino/operator/exchange/LocalExchange.java

```diff
@@ -141,7 +143,12 @@ else if (isScaledWriterHashDistribution(partitioning)) {
                     bufferCount,
                     1,
                     writerScalingMinDataProcessed.toBytes(),
-                    getSkewedPartitionMinDataProcessedRebalanceThreshold(session).toBytes());
+                    getSkewedPartitionMinDataProcessedRebalanceThreshold(session).toBytes(),
+                    // Keep the maxPartitionsToRebalance to atleast writer count such that single partition writes do
+                    // not suffer from skewness and can scale uniformly across all writers. Additionally, note that
+                    // maxWriterCount is calculated considering memory into account. So, it is safe to set the
```

Comment:
`So, it is safe to set the maxScaling to maximum number of writers.`

* `maxScaling -> maxPartitionsToRebalance`?

I don't quite understand how `maxWriterCount` relate to `maxPartitionsToRebalance`? Every rebalanced partition adds extra memory usage

Reply:
What I mean is the comment is not that helpful in understanding what is happening unless one knows the context. I kind of know the context, but still find the comment hard to interpret.